app/services/project_phases/reschedule_service.rb

```diff
@@ -71,19 +72,39 @@ def reschedule_phase_and_retrieve_next_start(phase, from)
     def calculate_date_range(from, duration:)
       days = working_days_from(from, count: duration)
 
-      [days.first.date, days.last.date] if days.length == duration
+      [days.first, days.last] if days.length == duration
+    end
+
+    def reschedule_partial_phase_and_retrieve_next_start(phase, from)
+      phase.start_date = calculate_start_date(from)
+      next_start_date = phase.start_date
+
+      if phase.finish_date?
+        phase.finish_date = calculate_finish_date(phase, from)
+        phase.duration = phase.calculate_duration
```

Comment:
I still think that it may make it confusing, as when editing one phase, another one can become complete, it can even happen when just activating dates or even changing working days